### PR TITLE
Add Xur info

### PR DESCRIPTION
**Pull request by bhollis**
Add a menu item for when Xur is in town. This brings up a panel with Xur's wares, how much everything costs, how many strange coins you have, and lets you show the item details popup plus compare against any version of exotics you might already have to see if there's a better roll.

This also includes a lot of refactoring to simplify stuff, but also to enable more vendor displays in the future.

I'd be happy for opinions on the presentation - I tried to put a bunch of info into a small space.

image
image
image

(Sorry I can't demo the "compare items" too impressively, since I only have one existing copy of these exotics. It works though!)

### CODE REVIEW

File: app/scripts/services/dimXurService.factory.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+(function() {
+  'use strict';
+
+  angular.module('dimApp')
+    .factory('dimXurService', XurService);
+
+  XurService.$inject = ['$rootScope', '$q', 'dimBungieService', 'dimItemDefinitions', 'dimStoreService', '$http'];
+
+  function XurService($rootScope, $q, dimBungieService, dimItemDefinitions, dimStoreService, $http) {
+    var xurTest = false; // set this to true when you want to test but Xur's not around
```
Comment (kyleshay) [inline app/scripts/services/dimXurService.factory.js]: :+1: loading mock data!!! love it!

**Comment by kyleshay** [comment]
awesome! compare feature is also really cool. 

ademo

**Comment by SunburnedGoose** [comment]
https://www.reddit.com/r/DestinyTheGame/comments/4eblc1/dim_v35_the_one_before_the_april_update/d1yss7r

**Comment by kyleshay** [comment]
Wish there was a way to link this location. Is there an image URL that updates via third party that is reliable that we could use?

**Comment by bhollis** [comment]
Anything I could improve, or should we ship as-is in the next update?

**Comment by kyleshay** [comment]
works great, i say ship it

**Pull request merged**